rosidl_typesupport_opensplice_cpp/include/rosidl_typesupport_opensplice_cpp/responder.hpp

```diff
@@ -250,9 +251,11 @@ class Responder
   noexcept
   {
     response.sequence_number_ = request_header.sequence_number;
-    response.client_guid_0_ = *(reinterpret_cast<const uint64_t *>(&request_header.writer_guid[0]));
-    response.client_guid_1_ = *(reinterpret_cast<const uint64_t *>(
-        &request_header.writer_guid[0] + sizeof(response.client_guid_0_)));
+    std::memcpy(
+      &response.client_guid_0_, &request_header.writer_guid[0], sizeof(response.client_guid_0_));
+    std::memcpy(
+      &response.client_guid_1_, &request_header.writer_guid[0] + sizeof(&response.client_guid_0_),
```

Comment:
`writer_guid` is defined as an array of 16 `int8_t` (https://github.com/ros2/rmw/blob/master/rmw/include/rmw/types.h#L100), so both suggestions would point to the second element of the array, instead of the eighth. I'm using `sizeof` instead of hardcoding the array element (8) because I think this looks slightly more readable.